Recruit-Training/Embedded-Recruit-Training/Core/Src/main.c

```diff
@@ -0,0 +1,458 @@
+    HAL_GPIO_WritePin(GPIOA, LED_RED_Pin | LED_GREEN_Pin | GPIO_PIN_7, GPIO_PIN_RESET);
+
+    /*Configure GPIO pins : LED_RED_Pin LED_GREEN_Pin PA7 */
+    GPIO_InitStruct.Pin = LED_RED_Pin | LED_GREEN_Pin | GPIO_PIN_7;
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+}
+
+/* USER CODE BEGIN 4 */
+void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
+{
+    CAN_RxHeaderTypeDef hdr;
+    uint8_t             data[8] = {0};
+
+    if (HAL_CAN_GetRxMessage(hcan, CAN_RX_FIFO0, &hdr, data) != HAL_OK)
+    {
+        return;
+    }
+
+    //TODO: Match StdId of header and data length content for green and blue messages and check data for set bit of toggling green and blue led
+    if (hdr.StdId == BLUE_MESSAGE_STDID && hdr.DLC == 1) 
+    {
+        uint8_t bit = 0b10001001
+
+        if(data[0] & bit == bit)  // & operator is  the AND gate to multiply 8-bit numbers
+        {
+            BlueToggleLED = 1;
+        }
+
+        else
+        {    
+            BlueToggleLED = 0;
+        }
+        
+    }
+
+    if(hdr.StdId == GREEN_MESSAGE_STDID && hdr.DLC == 1)
+    {
+        uint8_t mask = 0b00000011
+
+        if(data[0] & mask == mask)
+        {
+            GreenToggleLED = 1;
+        }
+       /* else
+        {
+            GreenToggleLED = 0;
+        }*/
+        
+    }
+}
+
+static void MX_CAN2_UserInit(void)
+{
+    CAN_FilterTypeDef blueMessageFilterConfig;
+    blueMessageFilterConfig.FilterBank = 0; // Use first filter bank
+    blueMessageFilterConfig.FilterMode = CAN_FILTERMODE_IDLIST; //Look for specific CAN messages
+    blueMessageFilterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+    blueMessageFilterConfig.FilterIdHigh = BLUE_MESSAGE_STDID << 5; // Filter registers need to be shifted left 5 bits
+    blueMessageFilterConfig.FilterIdLow = 0;
+    blueMessageFilterConfig.FilterMaskIdHigh = 0;
+    blueMessageFilterConfig.FilterFIFOAssignment = 0; //unused
+    blueMessageFilterConfig.FilterActivation = ENABLE;
+    blueMessageFilterConfig.SlaveStartFilterBank = 0; // set all filter banks for CAN2
+
+    if (HAL_CAN_ConfigFilter(&hcan2, &blueMessageFilterConfig) != HAL_OK)
+    {
+        /* Filter configuration Error */
+        Error_Handler();
+    }
+
+    //TODO: Configure filter for green message
+    CAN_FIlterTypeDef greenMessageFilterConfig;
+    greenMessageFilterConfig.FilterBank = 1;
+    greenMessageFilterConfig.FilterMode = CAN_FILTERMODE_IDLIST;
+    greenMessageFilterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+    greenMessageFilterConfig.FilterIdHigh = GREEN_MESSAGE_STDID;
+    greenMessageFilterConfig.FilterIdLow = 0;
+    greenMessageFilterConfig.FilterMaskIdHigh = 0;
+    greenMessageFilterConfig.FilterFIFOAssignment = 0;
+    greenMessageFilterConfig.FilterActivation = ENABLE;
+    greenMessageFilterConfig.SlaveStartFilterBank = 0;
+
```

Comment:
Need to call `HAL_CAN_ConfigFilter` just like was done above for `blueMessageFilterConfig`